server/routerlicious/packages/lambdas/src/deli/lambda.ts

```diff
@@ -639,15 +647,19 @@ export class DeliLambda extends TypedEventEmitter<IDeliLambdaEvents> implements
 		this.updateCheckpointMessages(rawMessage);
 
 		if (this.lastMessageType === MessageType.ClientJoin) {
-			this.globalCheckpointOnly = false;
+			this.recievedNoClientOp = false;
+			if (this.localCheckpointEnabled) {
+				this.globalCheckpointOnly = false;
+			}
+		} else if (this.lastMessageType === MessageType.NoClient) {
+			this.recievedNoClientOp = true;
+			if (this.localCheckpointEnabled) {
+				this.globalCheckpointOnly = true;
+			}
 		}
 
 		const checkpointReason = this.getCheckpointReason(this.lastMessageType);
 		if (checkpointReason !== undefined) {
-			// Set a flag to route all checkpoints to global collection if there are no active clients, and reset when clients join
```

Comment:
Can you bring back this comment? or add a few altered but similar ones?

Reply:
Good catch. I added the comment when `globalCheckpointOnly` is first initialized